src/frontend/src/index.ts

```diff
@@ -104,17 +105,24 @@ const init = async () => {
   const addDeviceAnchor = getAddDeviceAnchor();
   if (addDeviceAnchor !== undefined) {
     // Register this device (tentatively)
-    await registerTentativeDevice(addDeviceAnchor, connection);
-
-    const i18n = new I18n();
-    const copy = i18n.i18n(copyJson);
-
-    // Show a good bye message
-    showMessage({
-      message: copy.close_page_device_added,
-      role: "notify-device-added",
-    });
-    return;
+    const { userNumber, alias: deviceAlias } = await registerTentativeDevice(
+      addDeviceAnchor,
+      connection
+    );
+
+    // Display a success page once device added (above registerTentativeDevice **never** returns if it fails)
+    await addDeviceSuccess({ deviceAlias });
+
+    // If user "Click" continue in success page, proceed with authentication
+    const result = await authenticate(connection, userNumber);
+    const { success } = await handleLoginFlowResult(result);
+
+    // User have successfully signed-in we can jump to manage page
+    if (nonNullish(success)) {
+      const { connection: authenticatedConnection } = success;
+      await manage({ userNumber, authenticatedConnection });
```

Comment:
what do you think about this?

```suggestion
      return await manage({ userNumber, authenticatedConnection });
```

this would ensure that if `manage` ever returns anything meaningful, the typechecker would warn us about this call here (you can change `manage` to return `never`)

Reply:
I never do this and never had to but if you wish, no strong opinion.